builder/dispatchers.go

```diff
@@ -552,3 +605,53 @@ func volume(b *builder, args []string, attributes map[string]bool, original stri
 	}
 	return nil
 }
+
+// ARG --default=bar --description='my variable' foo
+//
+// Adds the variable foo to the trusted list of variables that can be passed
+// to builder using the --build-arg flag for expansion/subsitution or passing to 'run'.
+// Dockerfile author may optionally set a default value and description of this variable.
+func arg(b *builder, args []string, attributes map[string]bool, original string) error {
+	const (
+		def  = "default"
+		desc = "description"
+	)
+
+	if len(args) != 1 {
+		return fmt.Errorf("ARG requires exactly one argument")
+	}
+
+	// 'arg' accepts the default and description flags
+	b.BuilderFlags.AddString(def, "")
```

Comment:
this should be: 
```
flDefault := b.BuilderFlags.AddString(def, "")
```
and then you can use:
`flDefault` instead of `b.BuilderFlags.used[def]`  down below on line 640. "used" is an internal thing you shouldn't access.  Same for the flDescription.